### ISSUE TITLE
Part of Sentinel-2 images missing when atmospheric corrected

### ISSUE THREAD

**Issue by yukaribbba** [pytroll/satpy#1625]
**Describe the bug**
By now I have encountered this problem on 3 datasets. They're actually 2 scenes since one of them is cut off from the middle.
Here are the data files: [https://drive.google.com/drive/folders/1tVDX9EaGiCJUUz0glx1dUDiebks8KuES?usp=sharing](url)
**These images are created using 'marine_clean_aerosol' atm correction mode.**
![image](https://user-images.githubusercontent.com/72339781/112842495-30230500-90d4-11eb-96f5-7ab9e4f62e98.png)
![image](https://user-images.githubusercontent.com/72339781/112842530-3913d680-90d4-11eb-9ef8-d79d621843ce.png)
![image](https://user-images.githubusercontent.com/72339781/112842550-3fa24e00-90d4-11eb-8c67-8f1cb3a87d9e.png)

**After switching to 'raw' mode, which means no corrections, I got the completed ones.**
![image](https://user-images.githubusercontent.com/72339781/112842872-9740b980-90d4-11eb-9341-8f0248f3866b.png)
![image](https://user-images.githubusercontent.com/72339781/112842893-9dcf3100-90d4-11eb-9d8f-317b7f3cae5c.png)
![image](https://user-images.githubusercontent.com/72339781/112842923-a3c51200-90d4-11eb-96e7-cb9ead86a6f0.png)

**To Reproduce**

```python
#!/usr/bin/env python
from satpy import Scene, find_files_and_readers
files = find_files_and_readers(base_dir="/mnt/hgfs/Downloads/Sat/",
                               reader='msi_safe')
scn = Scene(filenames=files)
composite = 'true_color_marine_clean'
scn.load([composite])
scn.save_dataset(composite, filename='{name}_{start_time:%Y%m%d_%H%M%S}.tif', writer='geotiff', base_dir="/mnt/hgfs/Downloads/Sat/", num_threads=8, compress=None, fill_value=0)

```

**Expected behavior**
No missing parts.

**Actual results**
```
/home/ll/anaconda3/lib/python3.8/site-packages/satpy/readers/msi_safe.py:199: RuntimeWarning: Mean of empty slice
  return np.nanmean(np.dstack(arrays), -1)
/home/ll/anaconda3/lib/python3.8/site-packages/satpy/readers/msi_safe.py:199: RuntimeWarning: Mean of empty slice
  return np.nanmean(np.dstack(arrays), -1)
/home/ll/anaconda3/lib/python3.8/site-packages/pyproj/crs/crs.py:543: UserWarning: You will likely lose important projection information when converting to a PROJ string from another format. See: https://proj.org/faq.html#what-is-the-best-format-for-describing-coordinate-reference-systems
  proj_string = self.to_proj4()
/home/ll/anaconda3/lib/python3.8/site-packages/pyproj/crs/crs.py:543: UserWarning: You will likely lose important projection information when converting to a PROJ string from another format. See: https://proj.org/faq.html#what-is-the-best-format-for-describing-coordinate-reference-systems
  proj_string = self.to_proj4()
/home/ll/anaconda3/lib/python3.8/site-packages/pyproj/crs/crs.py:543: UserWarning: You will likely lose important projection information when converting to a PROJ string from another format. See: https://proj.org/faq.html#what-is-the-best-format-for-describing-coordinate-reference-systems
  proj_string = self.to_proj4()
/home/ll/anaconda3/lib/python3.8/site-packages/pyproj/crs/crs.py:543: UserWarning: You will likely lose important projection information when converting to a PROJ string from another format. See: https://proj.org/faq.html#what-is-the-best-format-for-describing-coordinate-reference-systems
  proj_string = self.to_proj4()
/home/ll/anaconda3/lib/python3.8/site-packages/dask/core.py:121: RuntimeWarning: invalid value encountered in log
  return func(*(_execute_task(a, cache) for a in args))

```

**Environment Info:**
 - OS: Ubuntu 20.04 64bit
 - Satpy Version: 0.27
 - RSR Data Version: 1.0.16
 - LUT version: 1.0.1

**Comment by mraspaud** [comment]
@yukaribbba thank you very much for reporting this issue.

I'm trying to access the data files, but they seem to be protected with an access code?

**Comment by yukaribbba** [comment]
@mraspaud Sorry I got messed with the sharing options. It is now open to anyone without code.

**Comment by mraspaud** [comment]
thanks!

**Comment by mraspaud** [comment]
@yukaribbba I dowloaded the first file and can't reproduce the error. I get a full image. Could you tell me what version of trollimage and pyspectral you have?

**Comment by yukaribbba** [comment]
@mraspaud 
pyspectral                0.10.4             pyhd8ed1ab_0    conda-forge
trollimage                1.15.0             pyhd8ed1ab_0    conda-forge

**Comment by mraspaud** [comment]
I think I found the problem: it looks like the satellite angles that are needed for the atmospheric correction aren't generated properly.
Here for example are the satellite zenith angles, and are invalid for half of the scene...
![Figure_1](https://user-images.githubusercontent.com/167802/113333389-a9b03280-9322-11eb-8a17-90f817a96024.png)

**Comment by mraspaud** [comment]
Ok, it looks like the angles array is getting transposed somehow during interpolation.

**Comment by mraspaud** [comment]
@yukaribbba I think I have a fix in #1628, could you tell me if it works for you?

**Comment by yukaribbba** [comment]
@mraspaud It works. Now I have full scenes under atm correction. Thank you!
![image](https://user-images.githubusercontent.com/72339781/113366780-cc773100-938c-11eb-94bd-b11b0910ff7c.png)
![image](https://user-images.githubusercontent.com/72339781/113366795-d26d1200-938c-11eb-9f87-f79e725c5633.png)
![image](https://user-images.githubusercontent.com/72339781/113366810-de58d400-938c-11eb-8fe9-b3877394b703.png)
